lib/src/map/map_builder.dart

```diff
@@ -53,6 +44,20 @@ class MapBuilder<K, V> {
     updates(this);
   }
 
+  /// Replaces all elements with elements from a [Map] or [BuiltMap].
+  void replace(map) {
+    if (map is BuiltMap<K, V>) {
+      _replaceFromBuiltMap(map);
+    } else if (map is BuiltMap) {
+      _replaceFromSafeMap(new Map<K, V>.from(map._map));
```

Comment:
The copy here is to set the generic type parameters.

Currently we partially rely on type checking for the underlying collections. We could switch to using dynamic underlying collections and doing all type checking explicitly ... in fact that's probably the smart thing to do, so we fully control type checking. Will add a note to the issue about type checking.